ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/base-journey/components/choice-textbox-form.ts

```diff
@@ -0,0 +1,42 @@
+import { ChoiceForm } from './choice-form';
+import { FormControl, Validators } from '@angular/forms';
+import { ValidateForWhiteSpace } from '../../shared/validators/whitespace-validator';
+
+export class ChoiceTextboxForm extends ChoiceForm {
+
+    readonly textInput = new FormControl('');
+
+    constructor() {
+      super();
+      this.formGroup.addControl('textInput', this.textInput);
+
+      this.choice.valueChanges.subscribe(value => {
+        if (value) {
+          // If the value is true, the text input is required
+          this.textInput.setValidators([Validators.required, ValidateForWhiteSpace]);
+          this.textInput.markAsUntouched();
+        } else {
+          this.textInput.clearValidators();
+        }
```

Comment:
Is it always we have text field only for 'Yes' choice answer? If we need text field for 'No'?

Reply:
That's a great question. I double checked on master,
about you for citizen = uses yes
about you (leg rep) = uses yes
about your client = uses yes
hearing suitability = uses yes

So yeah, they're all using yes right now. If they do end up using no at some point we can always extend this functionality.